jcl/src/java.base/share/classes/java/lang/ClassLoader.java

```diff
@@ -1982,10 +1999,49 @@ static void loadLibrary(Class<?> caller, String name, boolean fullPath) {
 
 /*[IF Java15]*/
 static void loadLibrary(Class<?> caller, File file) {
-	loadLibraryWithPath(file.getAbsolutePath(), caller.getClassLoaderImpl(), null);
+	ClassLoader loader = (caller == null) ? null : caller.getClassLoader();
+	NativeLibraries nls = (loader == null) ? bootstrapClassLoader.nativelibs : loader.nativelibs;
+	NativeLibrary nl = nls.loadLibrary(caller, file);
+	if (nl == null) {
+		throw new UnsatisfiedLinkError("Can't load " + file); //$NON-NLS-1$
```

Comment:
Why not NLS?

Reply:
Was lured away by https://github.com/eclipse/openj9/blob/f49490ab44a652e72d6d8f96b38b6095ce0cc97d/jcl/src/java.base/share/classes/java/lang/ClassLoader.java#L1970
Fixed it as well.

Please have another look.